registrations/signals.py

```diff
@@ -36,3 +64,31 @@ def signup_group_post_delete(
     sender: type[SignUpGroup], instance: SignUpGroup, **kwargs: dict
 ) -> None:
     _signup_or_group_post_delete(instance)
+
+
+@receiver(
+    pre_save,
+    sender=Event,
+    dispatch_uid="notify_signups_on_event_cancelled_pre_save",
+)
+def notify_signups_on_event_cancelled_pre_save(
+    sender: type[Event], instance: Event, **kwargs: dict
+) -> None:
+    if not (instance.pk and instance.event_status == Event.Status.CANCELLED):
+        return
+
+    old_instance = Event.objects.filter(pk=instance.pk).first()
+    if old_instance and old_instance.event_status != instance.event_status:
+        instance._send_event_cancellation_notification = True
+
+
+@receiver(
+    post_save,
+    sender=Event,
+    dispatch_uid="notify_signups_on_event_cancelled_post_save",
+)
+def notify_signups_on_event_cancelled_post_save(
+    sender: type[Event], instance: Event, **kwargs: dict
+) -> None:
+    if getattr(instance, "_send_event_cancellation_notification", False):
+        transaction.on_commit(lambda: _send_event_cancellation_notification(instance))
```

Comment:
Now that we have django-mailer, I think we want to bind notification sending (= actually adding notifications to be sent) exactly to event saving, and therefore `Event.save()` should be wrapped inside a transaction. And when that is the case, I think we can call `_send_event_cancellation_notification()` in the `pre_save` handler, because the actual sending will be then rolled back if the event saving fails. And then there is no need for this `post_save` handler at all. Maybe. 🤔